test/libsolidity/syntaxTests/operators/custom/implicit_conversion_is_blocked.sol

```diff
@@ -0,0 +1,34 @@
+using {add as +, unsub as -} for S;
+
+struct S {
+    uint x;
+}
+
+function add(S memory _a, S memory) returns (S memory) {
+    return _a;
+}
+
+function unsub(S memory _a) returns (S memory) {
+    return _a;
+}
+
+contract C {
+    S s;
+
+    function test() public {
+        S memory sTmp;
+        s + s;
+        sTmp + true;
+        true + s;
+        -sTmp;
+        -s;
```

Comment:
Are implicit conversions completely blocked or in just some cases? For example can a `calldata` argument be converted to `memory`? `memory` to `storage`? `storage` to `memory`? `int32` to `int64`?

Did we decide no conversion for now or should that work like in function calls?

In any case, please add test cases covering these if you don't have them yet.

Reply:
Smaller integer types are implicitly convertible to larger ones. For example this works:
```solidity
contract C {
    function f() public pure {
        uint16 x;
        uint32 y;
        y = x;
    }
}
```

There are also some other conversions for value types. E.g. `address payable` to `address`.

Reply:
Now, that I think of it - do we have any test for operators where the user-define type is an address?